scripts/build-and-provide-package

```diff
@@ -627,9 +627,10 @@ autopkgtest_results() {
 
 cowbuilder_run() {
   echo "*** cowbuilder build phase for arch $architecture ***"
-  mkdir -p "$WORKSPACE"/binaries/ /tmp/apt-$$
+  tmpaptdir=$(mktemp -d /tmp/apt-$(echo ${JOB_NAME} | sed -e 's,/,+,g')_XXXXXX)
```

Comment:
You're right, it could contain spaces, which would probably break things. Any other concerning characters?

Alternatively, I only wanted to make the directory name more obvious, but am happy to use mktemp only. Would that be preferred?